Cargo.toml

```diff
@@ -38,14 +38,17 @@ lazy_static = { version = "1.0.2", optional = true }
 thiserror = { version = "1.0.31", optional = true }
 smallvec = { version = "1.0", optional = true }
 
+# https://github.com/open-telemetry/opentelemetry-rust/issues/2165
+tonic = { version = "0.12.3", optional = true }
```

Comment:
In OpenTelemetry, [version 0.12 of `tonic` is specified](https://github.com/open-telemetry/opentelemetry-rust/blob/21fc16c0daa3ea8dbede23c814bda4f22d3dfef6/Cargo.toml#L43). However, if version 0.12.2 is used, the compilation fails.

Reply:
doesn't `cargo update` solve the issue? Seems somehow v0.12.2 is cached locally, and need to be cleaned.

Reply:
@lalitb Thank you for the suggestion. After using `cargo update` to update `tonic` from 0.12.2 to 0.12.3, the error was resolved.